### PR TITLE
use player monoids if available

### PR DESCRIPTION
**Pull request by fluxionary**
Currently, if a player has altered speed/jump/gravity from e.g. armor, and they sit/lay, they will lose the physics change when they get up. Using player monoids allows the proper values to be set when the player stands up.

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -1,15 +1,38 @@
+local function freeze(player)
```
Comment (SmallJoker) [inline init.lua:3]: You could also update `default.player_attached[name]` here. Both are always used together.

File: init.lua [hunk 2]
```diff
@@ -1,15 +1,38 @@
...
+                player_monoids.speed:add_change(player, 0, "cozy:speed")
+                player_monoids.jump:add_change(player, 0, "cozy:jump")
+                player_monoids.gravity:add_change(player, 0, "cozy:gravity")
+        else
+                player:set_physics_override({speed = 0, jump = 0, gravity = 0})
+        end
+end
+
+local function unfreeze(player)
+        if minetest.global_exists("player_monoids") then
```
Comment (SmallJoker) [inline init.lua]: Value can be cached. No need for double-checks (and neither in run-time).

File: init.lua (hunk 1 above)
Comment (fluxionary) [reply init.lua:3]: Refactored the code to pull in the lines that are always used together.

File: init.lua (hunk 2 above)
Comment (fluxionary) [reply init.lua]: Cached this value

**Comment by SmallJoker** [comment]
LGTM, will merge in a few days if there aren't any comments to handle.

PS: Please @ mention me in a week in case I forgot this PR.

**Comment by tuedel** [comment]
@SmallJoker Ping :)

**Comment by SmallJoker** [comment]
Oof, thanks.

**Pull request merged**